crates/store/re_remote_store_types/src/codec.rs

```diff
@@ -0,0 +1,253 @@
+use arrow2::error::Error as ArrowError;
+use arrow2::io::ipc::{read, write};
+use re_dataframe::TransportChunk;
+
+use crate::v0::EncoderVersion;
+
+#[derive(Debug, thiserror::Error)]
+pub enum CodecError {
+    #[error("Arrow serialization error: {0}")]
+    ArrowSerialization(ArrowError),
+
+    #[error("Failed to decode message header {0}")]
+    HeaderDecoding(std::io::Error),
+
+    #[error("Failed to encode message header {0}")]
+    HeaderEncoding(std::io::Error),
+
+    #[error("Missing record batch")]
+    MissingRecordBatch,
+
+    #[error("Unexpected stream state")]
+    UnexpectedStreamState,
+
+    #[error("Unknown message header")]
+    UnknownMessageHeader,
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Default)]
+pub struct MessageHader(pub u8);
+
+impl MessageHader {
+    pub const NO_DATA: Self = Self(0);
```

Comment:
I wouldn't use zero for anything but an error/unknown state, it's so easy for zeroes to slip in unnoticed...

Reply:
good point!